qualtran/bloqs/rotations/phase_gradient_test.py

```diff
@@ -37,12 +37,13 @@ def test_phase_gradient_state(n: int):
     state_prep_cirq_circuit = cirq.Circuit(
         cirq.H.on_each(*q), cirq.PhaseGradientGate(num_qubits=n, exponent=-1).on(*q)
     )
-    assert np.allclose(cirq.unitary(gate), cirq.unitary(state_prep_cirq_circuit))
-    assert np.allclose(
+    np.testing.assert_allclose(cirq.unitary(gate), cirq.unitary(state_prep_cirq_circuit))
+    np.testing.assert_allclose(
         cirq.unitary(gate**-1), cirq.unitary(cirq.inverse(state_prep_cirq_circuit))
     )
-    assert gate.t_complexity().rotations == n - 2
-    assert gate.t_complexity().clifford == n + 2
+    assert gate.t_complexity().t == 0
+    assert gate.t_complexity().rotations == n
+    assert gate.t_complexity().clifford == n
```

Comment:
@tanujkhattar is this right?

Reply:
No, this is incorrect. `PhaseGradientState(n)`  decomposes into $n$ Hadamards and $n$ Z rotations of the form $Z^{2^i}$. 2 out of those $n$ Z rotations are cliffords - $Z^1$ and $Z^0.5$ and remaining are non-clifford rotations.

Reply:
doing the suggested change leaves t=1, rot=n-3, clifford=n+2; which works by your logic if you consider z/4 to as a T